### PR TITLE
Thinks it is returning null when filtering out nulls in Laravel collection

### PR DESCRIPTION
**Pull request by kevin-schmitt**
Hello,

for this issue #130 

Thanks for review.

### CODE REVIEW

File: src/Codor/Sniffs/Files/ReturnNullSniff.php [hunk 1]
```diff
@@ -35,10 +35,16 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
         });
 
         $returnValueIndex = key($semicolons) - 1;
+        $comparaisonIndex = key($semicolons) - 3;
```
Comment (villfa) [inline src/Codor/Sniffs/Files/ReturnNullSniff.php]: Could replace `comparaison` by `comparison` everywhere and rename the files accordingly please?

File: src/Codor/Sniffs/Files/ReturnNullSniff.php [hunk 2]
```diff
@@ -35,10 +35,16 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
...
-        if ($scope[$returnValueIndex]['type'] === 'T_NULL') {
+        if ($scope[$returnValueIndex]['type'] === 'T_NULL' && $this->isComparaisonType($scope[$comparaisonIndex]['type']) === false) {
             $error = "Return null value found.";
             $phpcsFile->addError($error, $returnValueIndex, __CLASS__);
         }
     }
+
+    private function isComparaisonType(string $type) : bool
+    {
+        return (in_array($type, ['T_IS_NOT_IDENTICAL', 'T_IS_IDENTICAL']) == true);
```
Comment (villfa) [inline src/Codor/Sniffs/Files/ReturnNullSniff.php]:
```suggestion
        return in_array($type, ['T_IS_NOT_IDENTICAL', 'T_IS_IDENTICAL', 'T_IS_NOT_EQUAL', 'T_IS_EQUAL'], true);
```

**Review by villfa — CHANGES_REQUESTED** [review]
Thank you very much!
I just have a couple of remarks, but otherwise it looks good.

**Comment by kevin-schmitt** [comment]
Thank you very much for you help and review @villfa .
Can you check and validate if is good please?

**Review by villfa — APPROVED** [review]

**Comment by villfa** [comment]
Thanks :)

**Pull request merged**